### PR TITLE
add invoker function

### PR DESCRIPTION
**Pull request by andres-ml**

### CODE REVIEW

**Comment by ragboyjr** [comment]
Hmm, i think this is already supported via the `method` fn. Unfortunately `method` isn't documented/tested, but we could add that test/documentation, and I think that should suffice.

What do you think?

**Comment by andres-ml** [comment]
I actually saw it after the pull request but kind of dismissed it when I saw that the instance was taken as a 2nd parameter. I see now that its curried version `c\method` is effectively equal to `f\invoker`.
I kind of prefer `invoker` though; the curried versions of functions with variadic arguments feel kind of weird for me. Like, functions are data-last and in this case the data is the instance, but it goes before the method's arguments.

**Comment by ragboyjr** [comment]
hmm, i honestly don't see a difference between:

invoker main implementation

```php
function invoker(string $method, ...$args) {
     return function($instance) use($method, $args) {
         return call_user_func_array([$instance, $method], $args);
     };
 }
```

and the curried method


```php
function method($name, ...$optionalArgs)
{
    return function ($data) use($name, $optionalArgs) {
        return $data->{$name}(...$optionalArgs);
    };
}
```

They have the same api and functionality, just the internal variable names are different.

```php
invoker('methodName', $arg1, $arg2)($obj);
method('methodName', $arg1, $arg2)($obj);
```

**Comment by ragboyjr** [comment]
Unless i'm missing something, it doesn't seem to add value to have a new initially curried version of method. i'm not against an alias, but I'm also not convinced that invoker is a valid alias for what method is doing. Invoke implies invoking any type of callable, when in reality, it's ONLY invoking methods on objects. So the `method` naming that exists seems more appropriate.

I'd probably like to see the `method` function tested and therefore documented.

**Comment by andres-ml** [comment]
Yes, I mentioned they're effectively the same. It's just that (at least personally) it makes more sense to me for the `($method, ...$args)` header to be the default one; i.e. part of the non-curried version of the function. I don't really see a situation where I'd use `f\method($m, $o)` when I can do `$m->o()`, but I could potentially see myself using `c\invoker($m)`.
As for the naming, I [based it on Ramda](https://ramdajs.com/docs/#invoker), as I usually do.

**Comment by ragboyjr** [comment]
How would you feel about just making invoker an alias of curried method?

So the main definition is just

```php
function invoker(string $method, ...$args): callable {
  return \Krak\Fun\Curried\method($method, $args);
}
```

And we document method and invoker.

File: test/fn.spec.php [hunk 1]
```diff
@@ -964,6 +964,16 @@ function() { yield from [1,2,3]; },
...
     });
+    describe('invoker', function() {
+        docFn(invoker::class);
+        test("Builds a function that invokes a method with the specified parameters on its argument", function() {
+            $person1 = new \ArrayObject(['name' => 'John']);
+            $person2 = new \ArrayObject(['age' => '18']);
+            $isAgeDefined = invoker('offsetExists', 'age');
+            $res = arrayMap($isAgeDefined, [$person1, $person2]);
+            expect($res)->equal([false, true]);
+        });
```
Comment (ragboyjr) [inline test/fn.spec.php:975]: Just need a small documentation update mentioning that this is an alias of curried method and then we are good to go.

**Review by ragboyjr — CHANGES_REQUESTED** [review]

**Pull request closed without merging**